src/app/public/modules/colorpicker/colorpicker.component.html

```diff
@@ -1,186 +1,236 @@
 <div
   class="sky-input-group"
-  [ngClass]="{ 'sky-colorpicker-hidden': !isVisible }">
-  <ng-content>
-  </ng-content>
-  <sky-dropdown
-    [label]="'skyux_colorpicker_dropdown_button' | skyLibResources"
-    [messageStream]="dropdownController"
-    [dismissOnBlur]="false">
-    <sky-dropdown-menu>
-      <div class="sky-colorpicker-container">
+>
+  <div
+    [hidden]="true"
+  >
+    <ng-content></ng-content>
+  </div>
+
+  <button
+    aria-haspopup="true"
+    class="sky-colorpicker-button"
+    type="button"
+    [attr.aria-controls]="colorpickerId"
+    [attr.aria-expanded]="isOpen"
+    [attr.aria-label]="'skyux_colorpicker_dropdown_button' | skyLibResources"
+    [attr.id]="triggerButtonId"
+    [attr.title]="'skyux_colorpicker_dropdown_button' | skyLibResources"
+    [style.background-color]="backgroundColorForDisplay"
+    (click)="onButtonClick()"
+    #triggerButtonRef
+  ></button>
+
+  <ng-template #colorpickerTemplateRef>
+    <div
+      class="sky-colorpicker-container sky-shadow"
+      [attr.aria-labelledby]="triggerButtonId"
+      [attr.id]="colorpickerId"
+      [hidden]="!isVisible"
```

Comment:
Why do you have this `hidden` check here? Is there a case where you need to have the affixed item out there, but its innards are hidden?

Reply:
Yep! When the affixer can't find a valid placement, the picker should be hidden untiil the user scrolls the trigger button back into view. I'm subscribing to `affixer.placementChange` and setting `isVisible` to `false` if the placement equals `null`. The colorpicker isn't closed, it's just out of view, if that makes sense.

Reply:
Yeah that makes sense, and we'll probably need to do the same for other components (datepicker/timepicker).